aws/resources/lambda_layer.go

```diff
@@ -0,0 +1,143 @@
+package resources
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/report"
+	"github.com/gruntwork-io/cloud-nuke/telemetry"
+	commonTelemetry "github.com/gruntwork-io/go-commons/telemetry"
+)
+
+func (ll *LambdaLayers) getAll(c context.Context, configObj config.Config) ([]*string, error) {
+	var layers []*lambda.LayersListItem
+	var names []*string
+
+	err := ll.Client.ListLayersPages(
+		&lambda.ListLayersInput{}, func(page *lambda.ListLayersOutput, lastPage bool) bool {
+			for _, layer := range page.Layers {
+				logging.Logger.Debugf("Found layer! %s", layer)
+
+				if ll.shouldInclude(layer, configObj) {
+					layers = append(layers, layer)
+				}
+			}
+
+			return !lastPage
+		})
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, layer := range layers {
+		err := ll.Client.ListLayerVersionsPages(
+			&lambda.ListLayerVersionsInput{
+				LayerName: layer.LayerName,
+			}, func(page *lambda.ListLayerVersionsOutput, lastPage bool) bool {
+				for _, version := range page.LayerVersions {
+					logging.Logger.Debugf("Found layer version! %s", version)
+					names = append(names, layer.LayerName)
+				}
+
+				return !lastPage
+			})
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return names, nil
+}
+
+func (ll *LambdaLayers) shouldInclude(lambdaLayer *lambda.LayersListItem, configObj config.Config) bool {
+	if lambdaLayer == nil {
+		return false
+	}
+
+	// fnLastModified := aws.StringValue(lambdaLayer.LastModified)
+	fnLastModified := aws.StringValue(lambdaLayer.LatestMatchingVersion.CreatedDate)
```

Comment:
Lambda layers are immutable, so I think the created date of the latest version is on par with "last modified"? Open to suggestions though.

Reply:
SGTM. Can you add this information as a comment here?